feature-wallet-impl/src/main/java/jp/co/soramitsu/feature_wallet_impl/data/mappers/TransactionMappers.kt

```diff
@@ -1,102 +1,130 @@
-    Transaction.Status.PENDING -> TransactionLocal.Status.PENDING
-    Transaction.Status.COMPLETED -> TransactionLocal.Status.COMPLETED
-    Transaction.Status.FAILED -> TransactionLocal.Status.FAILED
-}
 
-fun mapTransactionStatusLocalToTransactionStatus(status: TransactionLocal.Status) = when (status) {
-    TransactionLocal.Status.PENDING -> Transaction.Status.PENDING
-    TransactionLocal.Status.COMPLETED -> Transaction.Status.COMPLETED
-    TransactionLocal.Status.FAILED -> Transaction.Status.FAILED
+fun mapOperationStatusToOperationLocalStatus(status: Operation.Status) = when (status) {
+    Operation.Status.PENDING -> OperationLocal.Status.PENDING
+    Operation.Status.COMPLETED -> OperationLocal.Status.COMPLETED
+    Operation.Status.FAILED -> OperationLocal.Status.FAILED
 }
 
-fun mapTransactionToTransactionModel(transaction: Transaction): TransactionModel {
-    return with(transaction) {
-        TransactionModel(
+fun mapOperationToOperationLocalDb(operation: Operation, source: OperationLocal.Source): OperationLocal {
+    with(operation) {
+        val amount = this.operation.getOperationAmount()
+        val fee = this.operation.getOperationFee()
+
+        return OperationLocal(
             hash = hash,
-            type = tokenType,
-            senderAddress = senderAddress,
-            recipientAddress = recipientAddress,
-            accountName = accountName,
-            isIncome = isIncome,
-            date = date,
-            amount = amount,
-            status = status,
-            fee = fee,
-            total = total
+            address = address,
+            time = time * 1000,
+            tokenType = mapTokenTypeToTokenTypeLocal(tokenType),
+            type = this.operation.header,
+            call = this.operation.subheader,
+            amount = amount?.toBigInteger(),
+            sender = (this.operation as? Operation.Operation.Transfer)?.sender,
+            receiver = (this.operation as? Operation.Operation.Transfer)?.receiver,
+            fee = fee?.toBigInteger(),
+            isReward = (this.operation as? Operation.Operation.Reward)?.isReward,
+            era = (this.operation as? Operation.Operation.Reward)?.era,
+            validator = (this.operation as? Operation.Operation.Reward)?.validator,
+            success = (this.operation as? Operation.Operation.Extrinsic)?.success,
+            status = mapOperationStatusToOperationLocalStatus(operation.operation.status),
+            source = source
         )
     }
 }
 
-fun mapTransactionLocalToTransaction(transactionLocal: TransactionLocal, accountName: String?): Transaction {
-    val tokenType = mapTokenTypeLocalToTokenType(transactionLocal.token)
+fun mapOperationLocalToOperation(operationLocal: OperationLocal, accountName: String?): Operation {
+    with(operationLocal) {
+        val operation = if (type != null && call != null && call != "Staking") {
```

Comment:
`type != null` -  how can it be null?

Reply:
type = header of Operation. And for transfer it is null

Reply:
I though type should be one of "Reward", "Trasnfer", "Extrinsic" to identify type of operation on data -> domain mapping

Reply:
Done